MovieQuiz/Services/QuestionFactory.swift

```diff
@@ -2,64 +2,104 @@ import Foundation
 
 class QuestionFactory: QuestionFactoryProtocol {
     weak var delegate: QuestionFactoryDelegate?
+    private let moviesLoader: MoviesLoading
+    private let conditions = ["больше", "меньше"]
     
-    init(delegate: QuestionFactoryDelegate?) {
+    init(delegate: QuestionFactoryDelegate?, moviesLoader: MoviesLoader) {
         self.delegate = delegate
+        self.moviesLoader = moviesLoader
     }
     
     func requestNextQuestion() {
-        let index = (0..<questions.count).randomElement() ?? 0
-        let question = questions[safe: index]
-        delegate?.didRecieveNextQuestion(question: question)
+        DispatchQueue.global().async { [weak self] in
+            guard let self = self else { return }
+            let index = (0..<self.movies.count).randomElement() ?? 0
+            
+            guard let movie = self.movies[safe: index] else { return }
+            var imageData = Data()
+
+            do {
+                imageData = try Data(contentsOf: movie.resizedImageURL)
+            } catch {
+                print("Failed to load image")
+            }
+
+            let rating = Float(movie.rating) ?? 0
+
+            let randomNumber = Int.random(in: 6...8)
+            let randomCondition = self.conditions.randomElement()!
```

Comment:
Все же старайтесь не использовать force unwrap'ы. Лучше дать дефолтный параметр:
```suggestion
            let randomCondition = self.conditions.randomElement() ?? "больше"
```

Reply:
@ABashkirova Привет! Кажется, вы забыли оставить комментарии в моём новом PR 😅
https://github.com/kkkyifut/MovieQuiz-ios/pull/5